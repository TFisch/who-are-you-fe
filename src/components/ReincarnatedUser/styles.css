.table-headers {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  margin: 3rem 4rem auto 4rem;
  text-align: left;
  align-content: center;
  align-items: center;
  border: 0.1rem solid black;
  background-color: rgba(200, 244, 244, 0.8); }

.reincarnated-user {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  margin: auto 4rem;
  text-align: left;
  align-content: center;
  align-items: center;
  border-bottom: 0.1rem solid black;
  border-right: 0.1rem solid black;
  border-left: 0.1rem solid black;
  background-color: rgba(235, 244, 244, 0.8); }

.user-details {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  margin: auto 4rem;
  text-align: left;
  align-content: center;
  align-items: center;
  width: 100%; }

.user-delete-button {
  border-radius: 0.7rem;
  height: 2rem;
  align-content: center;
  align-items: center;
  margin: auto 3rem;
  width: 6rem;
  font-size: 0.9rem;
  cursor: pointer;
  border: 0.05rem solid; }

.user-delete-button:hover {
  color: #d8a1af;
  background-color: #3b1920; }

.user-notes {
  display: flex;
  flex-direction: row;
  justify-content: center;
  margin-right: 2rem; }

.user-notes:focus {
  outline: none; }

.submit-notes {
  border-radius: 0.7rem;
  height: 2rem;
  align-content: center;
  align-items: center;
  margin: auto;
  width: 7rem;
  font-size: 0.9rem;
  cursor: pointer;
  border: 0.05rem solid; }

.submit-notes:hover {
  color: #d8a1af;
  background-color: #3b1920; }
