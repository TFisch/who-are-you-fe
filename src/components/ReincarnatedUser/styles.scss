.table-headers {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: auto 4rem;
  text-align: left;
  align-content: center;
  align-items: center;
  border-bottom: 0.1rem solid black;
}

.reincarnated-user {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin: auto 4rem;
  text-align: left;
  align-content: center;
  align-items: center;
  border-bottom: 0.1rem solid black;
}

.user-details {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: auto 4rem;
  text-align: left;
  align-content: center;
  align-items: center;
  width: 100%;
}

.user-delete-button {
  border-radius: 10%;
  height: 2rem;
  align-content: center;
  align-items: center;
  margin: auto 0;
}

.user-notes {
  display: flex;
  flex-direction: row;
  justify-content: center;
  margin-right: 2rem;
}

.user-notes:focus {
  outline: none;
}

.submit-notes {
  border-radius: 10%;
  height: 2rem;
  align-content: center;
  align-items: center;
  margin: auto auto;
}
