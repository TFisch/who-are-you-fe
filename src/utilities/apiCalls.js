export const fetchDeaths = async () => {
  const url = process.env.REACT_APP_DATABASE_API_URL + '/api/v1/deaths';
  const response = await fetch(url);
  const data = await response.json();
  await console.log(data);

};

export const fetchDateId = async day => {
  const underScoredDay = day.replace(/ /, '_');
  const url =
    process.env.REACT_APP_DATABASE_API_URL + `/api/v1/dates/${underScoredDay}`;
  const response = await fetch(url);
  const dateId = await response.json();
  return await dateId;
};

export const fetchDeathByDate = async (dateId, year) => {
  const url =
    process.env.REACT_APP_DATABASE_API_URL + `/api/v1/deaths/${dateId}/${year}`;
  const response = await fetch(url);
  const dead = await response.json();
  return await dead;
};

export const fetchUsers = async () => {
  const url = process.env.REACT_APP_DATABASE_API_URL + `/api/v1/users/`;
  const response = await fetch(url);
  const users = await response.json();
  return await users;
};

export const postUsers = async (name, death, notes) => {
  console.log(death);
  const url = process.env.REACT_APP_DATABASE_API_URL + `/api/v1/users/`;
  const response = await fetch(url, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({
      name: name,
      death_id: death[0].id,
      date_id: death[0].date_id,
      notes
    })
  });
  const user = await response.json();
  return await user;
};

export const deleteUsers = async userId => {
  const url =
    process.env.REACT_APP_DATABASE_API_URL + `/api/v1/users/${userId}`;
  const response = await fetch(url, {
    method: 'DELETE'
  });
  return await response.json();
};

export const updateUser = async (userId, notes) => {
  const url =
    process.env.REACT_APP_DATABASE_API_URL + `/api/v1/users/${userId}`;
  const response = await fetch(url, {
    method: 'PATCH',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({
      notes
    })
  });
  return await response.json();
};
