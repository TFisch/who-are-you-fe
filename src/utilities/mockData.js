export const mockDeath = [{
  created_at: "2018-10-13T00:30:37.227Z",
  date_id: 414,
  deletable: false,
  id: 1362,
  person_name: "Lefty Gomez",
  updated_at: "2018-10-13T00:30:37.227Z",
  year: 1989
}]

export const mockUsers = [{
  astrologySign: "Aquarius",
  day: "FEBRUARY 17",
  id: 6,
  name: "Cody Taft",
  notes: "Well hello there",
  reincarnatedPerson: "Lefty Gomez",
  year: 1989
}]