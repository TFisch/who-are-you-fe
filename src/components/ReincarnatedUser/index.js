import React from 'react';
import './styles.css';

const ReincarnatedUser = ({ user, handleClick, submitNotes }) => {
  return (
    <div className="reincarnated-user" id={user.id}>
      <div className="user-details">
        <h2 className="user-name">{user.name}</h2>
        <h2 className="reincarnated-person">{user.reincarnatedPerson}</h2>
        <h2 className="day">{user.day}</h2>
        <h2 className="year">{user.year}</h2>
        <h2 className="astrology-sign">{user.astrologySign}</h2>
        <button
          className="user-delete-button"
          onClick={handleClick}
          id={user.id}
        >
          Delete User
        </button>
      </div>
      <div className="user-notes">
        <p contentEditable="true" className="user-notes">
          {user.notes}
        </p>
        <button className="submit-notes" onClick={submitNotes}>
          Submit Notes
        </button>
      </div>
    </div>
  );
};

export default ReincarnatedUser;
