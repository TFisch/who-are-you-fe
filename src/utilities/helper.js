import { fetchDateById, fetchDeathById } from './apiCalls';

export const getMonthString = month => {
  let cleanMonth;
  switch (month) {
    case '01':
      cleanMonth = 'JANUARY';
      break;
    case '02':
      cleanMonth = 'FEBRUARY';
      break;
    case '03':
      cleanMonth = 'MARCH';
      break;
    case '04':
      cleanMonth = 'APRIL';
      break;
    case '05':
      cleanMonth = 'MAY';
      break;
    case '06':
      cleanMonth = 'JUNE';
      break;
    case '07':
      cleanMonth = 'JULY';
      break;
    case '08':
      cleanMonth = 'AUGUST';
      break;
    case '09':
      cleanMonth = 'SEPTEMBER';
      break;
    case '10':
      cleanMonth = 'OCTOBER';
      break;
    case '11':
      cleanMonth = 'NOVEMBER';
      break;
    case '12':
      cleanMonth = 'DECEMBER';
      break;
    default:
      return;
  }
  return cleanMonth;
};

export const concatDates = (month, day) => {
  const cleanDate = month.concat(' ', day);
  return cleanDate;
};

export const checkDay = day => {
  let cleanDay = day;
  switch (day) {
    case '01':
      cleanDay = '1';
      break;
    case '02':
      cleanDay = '2';
      break;
    case '03':
      cleanDay = '3';
      break;
    case '04':
      cleanDay = '4';
      break;
    case '05':
      cleanDay = '5';
      break;
    case '06':
      cleanDay = '6';
      break;
    case '07':
      cleanDay = '7';
      break;
    case '08':
      cleanDay = '8';
      break;
    case '09':
      cleanDay = '9';
      break;
    default:
      return day;
  }
  return cleanDay;
};

export const userCleaner = async user => {
  const day = await fetchDateById(user.date_id);
  const deadPerson = await fetchDeathById(user.death_id);
  const cleanUser = await {
    id: user.id,
    name: user.name,
    day: day[0].day,
    reincarnatedPerson: deadPerson[0].person_name,
    year: deadPerson[0].year,
    notes: user.notes,
    astrologySign: day[0].astrology_sign
  };
  return await cleanUser;
};
