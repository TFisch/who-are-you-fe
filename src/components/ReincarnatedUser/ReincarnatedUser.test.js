import React from 'react';
import { shallow } from 'enzyme';
import ReincarnatedUser from './';
import { mockUser } from '../../utilities/mockData'

describe('ReincarnatedUser', () => {
  let wrapper;

  beforeEach(() => {
    const user = mockUser;
    wrapper = shallow(<ReincarnatedUser user={user} />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});