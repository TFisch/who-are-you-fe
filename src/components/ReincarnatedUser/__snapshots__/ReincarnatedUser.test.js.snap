// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReincarnatedUser should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReincarnatedUser
    user={
      Object {
        "astrologySign": "Aquarius",
        "day": "FEBRUARY 17",
        "id": 6,
        "name": "Cody Taft",
        "notes": "Well hello there",
        "reincarnatedPerson": "Lefty Gomez",
        "year": 1989,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="user-name"
        >
          Cody Taft
        </h3>,
        <h3
          className="reincarnated-person"
        >
          Lefty Gomez
        </h3>,
        <p
          className="day"
        >
          FEBRUARY 17
        </p>,
        <p
          className="year"
        >
          1989
        </p>,
        <p
          className="astrology-sign"
        >
          Aquarius
        </p>,
        <button
          className="user-delete-button"
          id={6}
        >
          Delete User
        </button>,
      ],
      "className": "reincarnated-user",
      "id": 6,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Cody Taft",
          "className": "user-name",
        },
        "ref": null,
        "rendered": "Cody Taft",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lefty Gomez",
          "className": "reincarnated-person",
        },
        "ref": null,
        "rendered": "Lefty Gomez",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "FEBRUARY 17",
          "className": "day",
        },
        "ref": null,
        "rendered": "FEBRUARY 17",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 1989,
          "className": "year",
        },
        "ref": null,
        "rendered": 1989,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Aquarius",
          "className": "astrology-sign",
        },
        "ref": null,
        "rendered": "Aquarius",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Delete User",
          "className": "user-delete-button",
          "id": 6,
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Delete User",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="user-name"
          >
            Cody Taft
          </h3>,
          <h3
            className="reincarnated-person"
          >
            Lefty Gomez
          </h3>,
          <p
            className="day"
          >
            FEBRUARY 17
          </p>,
          <p
            className="year"
          >
            1989
          </p>,
          <p
            className="astrology-sign"
          >
            Aquarius
          </p>,
          <button
            className="user-delete-button"
            id={6}
          >
            Delete User
          </button>,
        ],
        "className": "reincarnated-user",
        "id": 6,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cody Taft",
            "className": "user-name",
          },
          "ref": null,
          "rendered": "Cody Taft",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lefty Gomez",
            "className": "reincarnated-person",
          },
          "ref": null,
          "rendered": "Lefty Gomez",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "FEBRUARY 17",
            "className": "day",
          },
          "ref": null,
          "rendered": "FEBRUARY 17",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 1989,
            "className": "year",
          },
          "ref": null,
          "rendered": 1989,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Aquarius",
            "className": "astrology-sign",
          },
          "ref": null,
          "rendered": "Aquarius",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Delete User",
            "className": "user-delete-button",
            "id": 6,
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Delete User",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
